### PR TITLE
feat(saft): use saft identifiers instead of indices

### LINKED ISSUE #250: Keep track of the total SAFT NAV
## Issue summary

- The NAV of a SAFT is determined once a new record is added and remains static for the time being. SAFT's don't have price feeds yet. The NAV can only be changed using `SAFT::registry::report_nav`
- it is therefore difficult to distinguish between SAFT and liquid assets. however, since SAFT are purely static, the total value could be tracked quite easily as additional aggregated value within the `SAFTRegistry::ActiveSaft` map

## Other information and links

-

### LINKED ISSUE #256: Keep SAFT record indices consistent
## Issue summary

- single SAFT are identified by their index in the list of `SAFTRecords` in the `ActiveSAFTs` storage.
- SAFTs can be removed entirely; currently, this removes the `SAFTRecord` from the vec, shrinking the vec.
- Therefore an `index` emitted via the `SAFTAdded` event  may no longer correspond to the record.

## Other information and links

- Ref #250

### PR DESCRIPTION
**Pull request by mattsse**
## Changes

- introduce unique and constant identifiers for SAFT
- track total NAV of all SAFTs for an asset
- update benchmarks and tests

builds on #255 

## Tests

```
cargo t --all-features
```

## Issues

- Closes #250 
- Closes #256

### CODE REVIEW

**Review by clearloop — COMMENTED** [review]
Conflicts after #255

**Comment by mattsse** [comment]
> Conflicts after #255

merged main

File: primitives/primitives/src/types.rs [hunk 1]
```diff
@@ -56,4 +56,7 @@ pub type Signature = MultiSignature;
 /// Value type for price feeds.
 pub type Value = u128;
 
+/// Identifier for a SAFT
+pub type SAFTId = u32;
```
Comment (clearloop) [inline primitives/primitives/src/types.rs:60]: New type missed in `types.json`

**Review by clearloop — APPROVED** [review]

File: primitives/primitives/src/types.rs (hunk 1 above)
Comment (mattsse) [reply primitives/primitives/src/types.rs:60]: Good catch,
Can we add this to the checklist aswell, I've got a feeling that this will happen a lot more

Comment (clearloop) [reply primitives/primitives/src/types.rs:60]: The `e2e` tests check this automatically =.=

**Pull request merged**